README.md

```diff
@@ -52,13 +52,33 @@ Open `http://localhost:3000` with your browser.
 
 ## Options
 
+### `port`
+
 ```ts
 serve({
   fetch: app.fetch,
   port: 8787, // Port number, default is 3000
 })
 ```
 
+### `createServer`
+
+```ts
+import { createServer } from 'node:https'
+import fs from 'node:fs'
+
+//...
+
+serve({
+  fetch: app.fetch,
+  createServer: createServer,
+  serverOptions: {
```

Comment:
I'd suggest leaving the `serverOptions` out by simply using the createServer as a function. Not seen the `serverOption` documented or a similar pattern used by other adaptors. Could potentially be a breaking change tho.

```ts
serve({
  fetch: app.fetch,
  createServer: (options = {}) => {
    return createServer({ 
        ...options, 
        key: fs.readFileSync('test/fixtures/keys/agent1-key.pem'),
        cert: fs.readFileSync('test/fixtures/keys/agent1-cert.pem'),
      })
    },
})
```

Reply:
Okay! I think you are right.

Reply:
Ah, if we don't use `serverOptions`, we have to pass `requestListener` to createServer as an argument:

```ts
  const server = createAdaptorServer({
    fetch: app.fetch,
    createServer: (options = {}, requestListener: RequestListener) => {
      return createServerHTTPS(
        {
          ...options,
          key: fs.readFileSync('test/fixtures/keys/agent1-key.pem'),
          cert: fs.readFileSync('test/fixtures/keys/agent1-cert.pem'),
        },
        requestListener // <---
      )
    },
  })
```

It's a little bit annoying, so, keep using `serverOptions`.